src/main/java/app/komunumo/util/SecurityUtil.java

```diff
@@ -85,6 +87,27 @@ public static boolean hasRole(final @NotNull UserRole role) {
         return false;
     }
 
+    /**
+     * <p>Checks if the current user is of the provided type.</p>
+     *
+     * <p>The method checks if the authenticated user principal is of the type that is provided to the method.</p>
+     *
+     * @param type the type to check the user principal against (e.g., {@link UserType#LOCAL})
+     * @return {@code true} if the current user is of the type, else return {@code false}
+     */
+    public static boolean isUserType(final @NotNull UserType type) {
+        final var auth = SecurityContextHolder.getContext().getAuthentication();
+
+        if (auth != null && auth.isAuthenticated()
+                && auth.getPrincipal() instanceof UserPrincipal userPrincipal) {
+
+            final var typeName = type.name();
+            return Objects.equals(userPrincipal.getType(), typeName);
```

Comment:
When `type` is refactored to an enum, you can just do `type.equals(userPrincipal.getType())`

Reply:
Refactored to enum in UserPrinciple.